src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java

```diff
@@ -174,7 +183,7 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
                         filterForm.setStatus(condition.getVulnerabilityStatuses());
                     }
                     VulnerabilityTrendHelper.logMessage(listener, "filterForm: " + filterForm);
-                    if (queryBy == Constants.QUERY_BY_START_DATE) {
+                    if (queryBy == Constants.QUERY_BY_START_DATE || queryBy == Constants.QUERY_BY_PARAMETER) {
```

Comment:
Why do we need to add QUERY_BY_PARAMETER to this clause? Seems like if QUERY_BY_PARAMETER needs it, then the other methods will as well.

Reply:
Good eye! That's necessary to filter the results to a single application without having to include the application ID in the APPVERSIONTAG value.

With this clause, the request is:
`GET /Contrast/api/ng/[organization ID]/traces/[app ID]/filter/?appVersionTags=[APPVERSIONTAG]`

Without the clause, it would be:
`GET /Contrast/api/ng/[organization ID]/orgtraces/filter/?appVersionTags=[APPVERSIONTAG]`

In the latter case, all traces for all applications in that org with matching appVersionTags would be returned. I imagine this could lead to unexpected results with values like `1.0`.